### ISSUE TITLE
Better logging

### ISSUE THREAD

**Issue by nvanheuverzwijn** [ory/oathkeeper#227]
**Describe the bug**

With a JWT authenticators, verifying a JWT token without a `kid` output an obscure logs that requires the operator to read the code.

**Reproducing the bug**

1. Setup oathkeeper with a jwt authenticator
2. generator a jwt token without a `kid` header claim
3. send a request with the token to the protected endpoint

*Server logs*

time="2019-07-23T15:29:07Z" level=warning msg="The authentication handler encountered an error" access_url="http://auth-accp.kronos-web.com/api/test" authentication_handler=jwt error="Access credentials are not sufficient to access this resource" granted=false reason_id=authentication_handler_error

**Expected behavior**

A log that tells us that the kid is missing in the received token.
time="2019-07-23T15:29:07Z" level=warning msg="The authentication handler encountered an error" access_url="http://auth-accp.kronos-web.com/api/test" authentication_handler=jwt error="Access credentials are not sufficient to access this resource" granted=false reason_id=authentication_handler_error reason=The JSON Web Token must contain a kid header value but did not.

**Environment**

* v0.17.0-beta.1
* Environment: Debian docker kubernetes

**Additional context**

Log level is debug

**Comment by aeneasr** [comment]
Yeah that makes sense - and it should be documented. Would you be up for a PR? :)

/label enhancement

**Comment by nvanheuverzwijn** [comment]
@aeneasr Thanksf or your reply. I would like to contribute with a PR but this is the most I can do for now.

The problem starts from `credentials/verifier_default.go` line `45`. The reason of the error seems to be destroyed here : `pipeline/authn/authenticator_jwt.go`  line `99`.

I hope this can do a difference.

**Comment by aeneasr** [comment]
Incorrect stalebot detection - this was assigned a milestone.
